### ISSUE TITLE
Question about weight_decay interaction with fraction

### ISSUE THREAD

**Issue by eric-tc-wong** [microsoft/dion#120]
Hello,

Thanks for making this wonderful work available.

I have a question regarding the interaction between `weight_decay`  and `ortho_fraction` in Dion3. `weight_decay` is applied to the whole weight matrix whereas `scatter_add_` is applied on the top_k. `configs/dion3_160m.yaml` and `configs/normuon_160m.yaml` kept `weight_decay` constant, but I want to understand whether there's a need to scale `weight_decay` for consistency between Dion3 with ortho_fraction<1 and NorMuon. Would the `weight_decay` need to be smaller when using a small `ortho_fraction` since rows are decayed more often than they are selected?

I intended to do hyperparameter tuning, but the recommendation is to use a larger `ortho_fraction` for smaller model, and a larger one for large models. Would the optimal hyperparameters shift with the change in `ortho_fraction` and what is your recommendation?

**Comment by JohnLangford** [comment]
The issue here seems real---perhaps ideally we'd have a different parameterization of things.    That also applies to the learning rate itself---we see that a larger learning rate is desirable when using a smaller fraction.   In terms of recommendations, the right thing to do with weight decay is a little complex in general---see for example: https://arxiv.org/abs/2506.02285 .   

At a fixed learning rate with a variable fraction I've typically just set the weight decay to a small value and not worried about it to much.  In terms of which fraction to use, a significant question is whether or not you see improved performance with a 1/4 fraction and twice the learning rate regardless of model size, as we saw here: https://arxiv.org/pdf/2608.11612 .

**Comment by eric-tc-wong** [comment]
Thank you for the recommendations and resources!
